test/e2e-secondary-network/framework.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	aggregatorclientset "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
+
+	crdclientset "antrea.io/antrea/pkg/client/clientset/versioned"
+	antreae2e "antrea.io/antrea/test/e2e"
+)
+
+type SNTestData struct {
+	e2eTestData        *antreae2e.TestData
+	kubeConfig         *restclient.Config
+	clientset          kubernetes.Interface
+	aggregatorClient   aggregatorclientset.Interface
+	crdClient          crdclientset.Interface
+	logsDirForTestCase string
+	testNamespace      string
+}
+
+func createDirectory(path string) error {
+	return os.Mkdir(path, 0700)
+}
+
+const (
+	busyboxImage = "projects.registry.vmware.com/antrea/busybox"
+)
+
+var SNtestData *SNTestData
+
+type PodIPs struct {
+	ipv4      *net.IP
+	ipv6      *net.IP
+	ipStrings []string
+}
+
+type podInfo struct {
+	name      string
+	os        string
+	nodeName  string
+	namespace string
+}
+
+type ClusterInfo struct {
+	controlPlaneNodeName string
+}
+
+var clusterInfo ClusterInfo
+
+func (data *SNTestData) setupLogDirectoryForTest(testName string) error {
+	path := filepath.Join(testOptions.logsExportDir, testName)
+	// remove directory if it already exists. This ensures that we start with an empty directory
+	_ = os.RemoveAll(path)
+	err := createDirectory(path)
+	if err != nil {
+		return err
+	}
+	data.logsDirForTestCase = path
+	return nil
+}
+
+func (data *SNTestData) createClient(kubeconfigPath string) error {
+	e2edata = &antreae2e.TestData{}
+	if err := e2edata.CreateClient(kubeconfigPath); err != nil {
+		log.Fatalf("Error when creating K8s ClientSet: %v", err)
+		return err
+	}
+	data.e2eTestData = e2edata
+	return nil
+}
+
+func (data *SNTestData) RunPingCommandFromTestPod(podInfo podInfo, ns string, targetPodIPs *PodIPs, ctrName string, count int, size int) error {
```

Comment:
why do we need to redefine these functions?
could you just export the existing function in the other e2e package?

Reply:
Sorry I missed to update on this comment. runPingCommandFromTestPod at test/e2e was not exported. exporting this would lead to changes all the other test functions which is using it already. If we are good with it, then great. I can goahead and make that change today and update the PR.

Reply:
took care of these changes couple of weeks back. sorry, missed to follow-up on this PR. Please review when you get sometime.